flow/raster_cache.cc

```diff
@@ -39,8 +46,21 @@ void RasterCacheResult::draw(DlCanvas& canvas, const DlPaint* paint) const {
              std::abs(bounds.height() - image_->dimensions().height()) <= 1);
   canvas.TransformReset();
   flow_.Step();
-  canvas.DrawImage(image_, {bounds.fLeft, bounds.fTop},
-                   DlImageSampling::kNearestNeighbor, paint);
+  if (is_root_canvas || !rtree_) {
+    canvas.DrawImage(image_, {bounds.fLeft, bounds.fTop},
+                     DlImageSampling::kNearestNeighbor, paint);
+  } else {
+    // On some platforms RTree from overlay layers is used for unobstructed
+    // platform views and hit testing. To preserve the RTree raster cache must
+    // paint individual rects instead of the whole image.
+    auto rects = rtree_->searchAndConsolidateRects(kGiantRect);
```

Comment:
It would be nice if we could quickly and easily estimate the density of the RTree and then just go through the single DrawImage above. Some (not sure of percentage) cache items will have a bg fill at the start.

Reply:
Maybe compare overall bounds (all RTree rectangles joined together) with area of largest rect? That would be pretty good indicator of a fill. It would require one pass through all rectangles (or gather this when constructing the RTree). But it should be able to detect fill.

Reply:
But you'd have to account for overdraw? The Region of the RTree might make a better source for that data.

Which reminds me, I was going to check if the RTree cached its Region or not.

Reply:
I'm not entirely sure what we're trying to save here. If the region is rectangular it will only yield one rectangle covering the entire area anyway.

Reply:
`RTree` doesn't cache the `Region`. There might be different region built depending on the query rect though so if we want to cache it we might need to do it per query rect.

Reply:
We could make a Region for the whole RTree before intersecting with the rect, though. We'd then need a Region::IntersectRect method, but that could run faster than the existing searchAndConsolidate. Could that amortize the Region work?

Food for thought...

Reply:
Getting a subregion by intersecting region with a rect is likely quite a lot cheaper than constructing region from rectangles. I created a new issue here since `DlRegion::MakeIntersection` is out of scope of this PR, though a version of it is implemented here: https://github.com/flutter/engine/pull/42620

For this though it would be best to have an intersection of region and rectangle, instead of two regions.